distributions/aws/examples/rds-s3/auto-rds-s3-setup.sh

```diff
@@ -0,0 +1,337 @@
+    associate_iam_oidc_provider
+    if [[ $? != 0 ]]; then
+        exit 1
+    fi
+    echo "OIDC Provider association done!"
+}
+
+associate_iam_oidc_provider() {
+    eksctl utils associate-iam-oidc-provider \
+        --region=$CLUSTER_REGION \
+        --cluster=$CLUSTER_NAME \
+        --approve
+}
+
+setup_iam_service_account() {
+    echo "Creating service account..."
+    create_iam_service_account
+    if [[ $? != 0 ]]; then
+        exit 1
+    fi
+    echo "Service account creation done!"
+}
+
+create_iam_service_account() {
+    eksctl create iamserviceaccount  \
+        --region $CLUSTER_REGION \
+        --name kubeflow-secrets-manager-sa  \
+        --namespace kubeflow  \
+        --cluster $CLUSTER_NAME \
+        --attach-policy-arn  arn:aws:iam::aws:policy/AmazonSSMReadOnlyAccess \
+        --attach-policy-arn arn:aws:iam::aws:policy/SecretsManagerReadWrite \
+        --override-existing-serviceaccounts \
+        --approve
+}
+
+setup_secrets_provider() {
+    echo "Installing secrets provider..."
+
+    install_secrets_store_csi_driver
+    if [[ $? != 0 ]]; then
+        exit 1
+    fi
+    
+    echo "Secrets provider install done!"
+}
+
+install_secrets_store_csi_driver() {
+    kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/v1.0.0/deploy/rbac-secretproviderclass.yaml \
+    && kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/v1.0.0/deploy/csidriver.yaml \
+    && kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/v1.0.0/deploy/secrets-store.csi.x-k8s.io_secretproviderclasses.yaml \
+    && kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/v1.0.0/deploy/secrets-store.csi.x-k8s.io_secretproviderclasspodstatuses.yaml \
+    && kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/v1.0.0/deploy/secrets-store-csi-driver.yaml \
+    && kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/v1.0.0/deploy/rbac-secretprovidersyncing.yaml \
+    && kubectl apply -f https://raw.githubusercontent.com/aws/secrets-store-csi-driver-provider-aws/main/deployment/aws-provider-installer.yaml
+}
+
+setup_kubeflow_pipelines() {
+    echo "================================================================="
+    echo "                  Kubeflow Pipelines Setup"
+    echo "================================================================="
+    
+    edit_pipelines_params_env
+}
+
+edit_pipelines_params_env(){
+
+    echo "Retrieving DB instance info..."
+    db_endpoint_url=$(aws rds describe-db-instances \
+        --region $CLUSTER_REGION \
+        --query "DBInstances[?DBInstanceIdentifier=='$DB_INSTANCE_NAME'].Endpoint.Address" | cut -d " " -f 2) \
+    && echo "Editing $PIPELINE_PARAMS_ENV_FILE" \
+    && sed -i 's/dbHost=.*'"/dbHost=$db_endpoint_url/" $PIPELINE_PARAMS_ENV_FILE \
```

Comment:
@AlexandreBrown I was trying to run the script and the setup worked smoothly but these sed commands are failing for me, could rerun and see if this still works for you ?
I am running this on my mac, what's about you ?

Reply:
@mbaijal I tested it numerous time on **linux** and it worked every time. 
I was not aware of this but there might be some differences between `sed` on linux vs `sed` on mac os (see  https://unix.stackexchange.com/questions/13711/differences-between-sed-on-mac-osx-and-other-standard-sed )

Could you confirm which command worked on mac os ?

Reply:
@mbaijal I will follow @surajkota 's idea mentioned [here](https://github.com/awslabs/kubeflow-manifests/pull/81) and convert this script to python.   
This should prevent any OS incompatibility.

Reply:
Thanks, that would be great !

Reply:
Done

Reply:
I'll review this tomorrow morning, thanks for the fast follows!